### PR TITLE
Dockerization

### PR DESCRIPTION
**Pull request by btruhand**
Dockerization of FISH

### CODE REVIEW

**Comment by jorgearanda** [comment]
Thanks for this PR again, @btruhand !

I have a question: we have as prerequisites ruby and sass. I don't see them included here, but maybe I'm missing something?

File: README.md [hunk 1]
```diff
@@ -35,6 +35,41 @@ These are located in the developer_scripts folder
...
+## Docker
+This project is also dockerized (no official repository available however). This project uses and tested on the following Docker technologies:
+
+* Docker for Mac:
+   * Client version: 17.03.1-ce
+   * Server version: 17.03.1-ce
+* Docker Compose version 3
+
+### Container
+The node application is run in a container named _dockered-fish_ and the MongoDB is run in a container
```
Comment (jorgearanda) [inline README.md]: Is it possible to make this run in containers simply named _fish_ and _mongo-fish_?

File: README.md [hunk 2]
```diff
@@ -35,6 +35,41 @@ These are located in the developer_scripts folder
...
+named _dockered-mongo-fish_. See `docker-compose.yml` for more details.
+
+### Usage
+In order to use the dockerized version do the following
+1. npm run build-docker
+2. Run the Docker services, one of
+   * `npm run start-docker`
+   * `npm run start-docker-detached` (detached docker mode)
```
Comment (jorgearanda) [inline README.md]: can you add a note or explain the difference between detached mode and default?

File: README.md [hunk 3]
```diff
@@ -35,6 +35,41 @@ These are located in the developer_scripts folder
...
+
+### Logs
+If Docker was run in non-detached mode then logs will be visible on console stdout.
+The node application and MongoDB logs may also be found in **fish.log** and **mongo.log** respectively.
+
+If Docker was run in detached mode you may see logs by running the following commands:
```
Comment (jorgearanda) [inline README.md]: Ah I see. Are these the only differences?

File: README.md (hunk 1 above)
Comment (btruhand) [inline README.md]: yeah definitely! I'll change it sometime this weekend

File: README.md (hunk 3 above)
Comment (btruhand) [inline README.md]: Essentially the difference is that with the ```-d``` flag it's run as a daemon and without it you will get the logs directly on your screen as you're running it as just a program and doing ```Ctrl+C``` will stop the containers

**Comment by btruhand** [comment]
Ah @jorgearanda about the Ruby and Sass, I didn't really think of putting it in the docker since I was thinking of providing the docker as a runtime container not necessarily a build container also. Would you like me to add in a build container for Ruby and Sass?

File: README.md (hunk 1 above)
Comment (jorgearanda) [inline README.md]: thanks!

File: README.md (hunk 3 above)
Comment (jorgearanda) [inline README.md]: OK, great, maybe we could call it daemon mode and standard mode?

**Comment by jorgearanda** [comment]
My knowledge of docker is still pretty limited—but what I was picturing was that the container would have all that is needed to run `fish` on a fresh box. Maybe that's a "build container"? If so, then yes, I think ruby and sass should be there as they are requisites (along with mongo)

File: README.md (hunk 3 above)
Comment (btruhand) [inline README.md]: Yeah that sounds a lot better

**Comment by btruhand** [comment]
That would constitute both a build and a runtime container I think. In my own understanding a build container is one used to build the dependencies and the runtime container is responsible to run it (assuming the dependencies are there already)

I'll make a diagram and put it into the ```README``` so it'll be clearer the structure of the current Docker setup.

- [ ] Make diagram of Docker setup

P.S Sorry for just replying, been quite busy myself. Will try to work on it this weekend :)

**Comment by jorgearanda** [comment]
You're awesome ;)

**Pull request merged**